ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/CommitContext.scala

```diff
@@ -47,17 +47,17 @@ private[kvutils] case class CommitContext(
   def preExecute: Boolean = recordTime.isEmpty
 
   /** Retrieve value from output state, or if not found, from input state. */
-  def get(key: DamlStateKey): Option[DamlStateValue] =
-    outputs.get(key).orElse {
-      val value = inputs.getOrElse(key, throw Err.MissingInputState(key))
-      accessedInputKeys += key
-      value
-    }
+  def get(key: DamlStateKey): Option[DamlStateValue] = {
+    val value = outputs.get(key).orElse(inputs.getOrElse(key, throw Err.MissingInputState(key)))
```

Comment:
This changes the logic to mark an output key as being accessed even if it was not declared in the input. Is this intended?

Reply:
There's the bug! Thank you for finding it. I was sure this was a pure refactoring.